### ISSUE TITLE
Playbook does not stop when package for unarchive is not available

### ISSUE THREAD

**Issue by stoeps13** [HCL-TECH-SOFTWARE/connections-automation#22]
The usage of unarchive with http seems not to fail when a package is not available, it fails several steps later, when the setup wants to access one of the unarchived files.

Can we create a role which testss all used packages if they are available on the http server? Any idea how to do this without downloading the files?

**Comment by dejo1307** [comment]
Yes, it is possible to fix it (I would say it is probably unintentionally ignored).

Which specific step/package is here the case? I can take a look now.

**Comment by stoeps13** [comment]
I had this on different steps, because I used the Flexnet filenames and haven't checked everything. So each time the unarchive is used, it tries to download and unarchive. When this fails because a file is missing, the playbook goes ahead to the next task. 

That's an issue on two sides. First it should already fail when it can't unarchive, that would be easier to troubleshoot and second the failure of missing packages appears after 10 - 80 minutes depending on the package. So a task which checks for all needed software packages could be run as a first task and immediately inform the Ansible user that something is missing or wrong.

I had this in DB2 and SDI for example.

So unarchive should fail, but also a check task would be awesome.

**Comment by dejo1307** [comment]
Ah, got it. Actually yes, can be a two step process (initially it was), to use get_url module to download the file and then to extract the file; in that case, if file can not be downloaded, it fails on spot. 

Let me try to figure out if I can use assert here to check if files exist on destination, that would make probably most sense (like: don't try to install for example TDI if TDI files can not be found and downloaded).

**Comment by dejo1307** [comment]
And thinking more about this, I guess I can use uri module for asserts to accomplish exactly this. Looking into it now.

**Comment by stoeps13** [comment]
Btw when you already investigating this. It seems the unarchive process is not idempotent, it downloads and extracts the packages on each run.
